frontend/components/dashboard/DashboardHeader.vue

```diff
@@ -87,13 +87,13 @@ const items = computed<MenuBarEntry[]>(() => {
     }
   }
   addToSortedItems($t('dashboard.header.validator'), dashboards.value?.validator_dashboards?.map((db) => {
-    const cd = db as CookieDashboard
-    return createMenuBarButton('validator', getDashboardName(cd), `${cd.hash !== undefined ? cd.hash : cd.id}`)
+    const gd = db as GuestDashboard
+    return createMenuBarButton('validator', getDashboardName(gd), `${gd.key !== undefined ? gd.key : gd.id}`)
```

Comment:
is this `typecasting` needed anyway?
```suggestion
    const guestDashboard = dashboard as GuestDashboard
    return createMenuBarButton('validator', getDashboardName(guestDashboard), `${guestDashboard.key !== undefined ? guestDashboard.key : guestDashboard.id}`)
```

Reply:
![image](https://github.com/user-attachments/assets/0c3fcc61-6651-429e-8189-5209c0256e72)
removing it causes TS errors, so without going too deep into this i'm saying yes

Reply:
haha welcome to the `typescript casting` hell 🔥